webpack.config.js

```diff
@@ -0,0 +1,220 @@
+          // plugins section. This plugin must be the last in order in order to
+          // work correctly (and thus needs to be on the top).
+          MiniCssExtractPlugin.loader,
+          {
+            loader: 'css-loader',
+            options: {
+              sourceMap: true,
+              // If modules is set to true _or_ it is an object, this enables
+              // CSS Modules - a means of un-globalizing CSS class selectors and
+              // letting us pretend that CSS files are actually standalone
+              // modules that don't interfere with other modules.
+              modules: {
+                exportLocalsConvention: 'camelCase',
+                // The localIdentName provides a template in which the class
+                // names are mangled by CSS Modules. By default it is just a
+                // random hash. Below, we use the file name ([name]) and the
+                // class name ([local]) along with part of the hash. This allows
+                // us to be able to see what classes are applied to what
+                // elements in a human readable way.
+                localIdentName: '[name]__[local]__[hash:base64:5]',
+              },
+              importLoaders: 1,
+            },
+          },
+          // PostCSS is akin to Babel.js, but for CSS files. PostCSS allows us
+          // to use cutting edge or experimental CSS features that browsers
+          // don't support yet, and it will convert that CSS into into CSS our
+          // browsers can understand. An example of such a feature is nested
+          // CSS. Each feature can be thought of as an individual plugin, listed
+          // below.
+          {
+            loader: 'postcss-loader',
+            options: {
+              sourceMap: true,
+              postcssOptions: {
+                plugins: [
+                  // Import one CSS from another.
+                  postcssImport(),
+                  // Use browser prefixes - a way for browsers to aggressively
+                  // support new CSS features. autoprefixer can be supplied with
+                  // a browser support matrix and autoprefixer will
+                  // automatically add prefixes _as needed_ to your CSS. See
+                  // browserlist (https://github.com/browserslist/browserslist)
+                  // for declaring such a matrix).
+                  autoprefixer(),
+                  // Use nested CSS. This lets you write selectors less
+                  // repetitiously.
+                  postcssNested(),
+                ],
+              },
+            },
+          },
+        ],
+      },
+      // Support the packaging of images.
+      {
+        test: /\.(jpeg|jpg|png|svg|gif)$/,
+        type: 'asset/resource',
+      },
+      // Support the packaging of fonts.
+      {
+        test: /\.(woff|woff2|eot|ttf|otf)$/i,
+        type: 'asset/resource',
+      },
+    ],
+  },
+};
```

Comment:
I feel like I learned more about webpack just going through this. Major kudos to you for pushing through and building your own repo!